[reproc] Build error

Package: reproc:x64-linux -> 14.2.4#1

**Host Environment**

- Host: x64-linux
- Compiler: GNU 13.2.0
-    vcpkg-tool version: 2023-10-18-27de5b69dac4b6fe8259d283cd4011e6d20a84ce
    vcpkg-scripts version: 8eb5735 2023-10-19 (5 days ago)

**To Reproduce**

`vcpkg install libsndfile[core,external-libs] kissfft fmt crossguid sdl2[x11] gl3w reproc gsl-lite concurrentqueue platform-folders catch2 --recurse`
**Failure logs**

```
-- Downloading https://github.com/DaanDeMeyer/reproc/archive/v14.2.4.tar.gz -> DaanDeMeyer-reproc-v14.2.4.tar.gz...
-- Extracting source /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/downloads/DaanDeMeyer-reproc-v14.2.4.tar.gz
-- Using source at /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean
-- Found external ninja('1.11.1').
-- Configuring x64-linux
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/cmake --build . --config Debug --target install -- -v -j5
    Working Directory: /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/x64-linux-dbg
    See logs for more information:
      /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/reproc/portfile.cmake:17 (vcpkg_cmake_install)
  scripts/ports.cmake:168 (include)



```
<details><summary>/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/install-x64-linux-dbg-out.log</summary>

```
Change Dir: '/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/x64-linux-dbg'

Run Build Command(s): /usr/bin/ninja -v -v -j5 install
[1/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/error.posix.c.o -MF reproc/CMakeFiles/reproc.dir/src/error.posix.c.o.d -o reproc/CMakeFiles/reproc.dir/src/error.posix.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/error.posix.c
[2/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/handle.posix.c.o -MF reproc/CMakeFiles/reproc.dir/src/handle.posix.c.o.d -o reproc/CMakeFiles/reproc.dir/src/handle.posix.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/handle.posix.c
[3/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/clock.posix.c.o -MF reproc/CMakeFiles/reproc.dir/src/clock.posix.c.o.d -o reproc/CMakeFiles/reproc.dir/src/clock.posix.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/clock.posix.c
[4/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/drain.c.o -MF reproc/CMakeFiles/reproc.dir/src/drain.c.o.d -o reproc/CMakeFiles/reproc.dir/src/drain.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/drain.c
[5/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/init.posix.c.o -MF reproc/CMakeFiles/reproc.dir/src/init.posix.c.o.d -o reproc/CMakeFiles/reproc.dir/src/init.posix.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/init.posix.c
[6/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/redirect.c.o -MF reproc/CMakeFiles/reproc.dir/src/redirect.c.o.d -o reproc/CMakeFiles/reproc.dir/src/redirect.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/redirect.c
[7/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/options.c.o -MF reproc/CMakeFiles/reproc.dir/src/options.c.o.d -o reproc/CMakeFiles/reproc.dir/src/options.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/options.c
[8/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/pipe.posix.c.o -MF reproc/CMakeFiles/reproc.dir/src/pipe.posix.c.o.d -o reproc/CMakeFiles/reproc.dir/src/pipe.posix.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/pipe.posix.c
[9/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/redirect.posix.c.o -MF reproc/CMakeFiles/reproc.dir/src/redirect.posix.c.o.d -o reproc/CMakeFiles/reproc.dir/src/redirect.posix.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/redirect.posix.c
[10/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/process.posix.c.o -MF reproc/CMakeFiles/reproc.dir/src/process.posix.c.o.d -o reproc/CMakeFiles/reproc.dir/src/process.posix.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/process.posix.c
[11/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/run.c.o -MF reproc/CMakeFiles/reproc.dir/src/run.c.o.d -o reproc/CMakeFiles/reproc.dir/src/run.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/run.c
[12/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/reproc.c.o -MF reproc/CMakeFiles/reproc.dir/src/reproc.c.o.d -o reproc/CMakeFiles/reproc.dir/src/reproc.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/reproc.c
[13/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/strv.c.o -MF reproc/CMakeFiles/reproc.dir/src/strv.c.o.d -o reproc/CMakeFiles/reproc.dir/src/strv.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/strv.c
[14/18] /usr/bin/cc -DREPROC_MULTITHREADED -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c99 -MD -MT reproc/CMakeFiles/reproc.dir/src/utf.posix.c.o -MF reproc/CMakeFiles/reproc.dir/src/utf.posix.c.o.d -o reproc/CMakeFiles/reproc.dir/src/utf.posix.c.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/src/utf.posix.c
[15/18] : && /usr/bin/cmake -E rm -f reproc/lib/libreproc.a && /usr/bin/ar qc reproc/lib/libreproc.a  reproc/CMakeFiles/reproc.dir/src/clock.posix.c.o reproc/CMakeFiles/reproc.dir/src/drain.c.o reproc/CMakeFiles/reproc.dir/src/error.posix.c.o reproc/CMakeFiles/reproc.dir/src/handle.posix.c.o reproc/CMakeFiles/reproc.dir/src/init.posix.c.o reproc/CMakeFiles/reproc.dir/src/options.c.o reproc/CMakeFiles/reproc.dir/src/pipe.posix.c.o reproc/CMakeFiles/reproc.dir/src/process.posix.c.o reproc/CMakeFiles/reproc.dir/src/redirect.posix.c.o reproc/CMakeFiles/reproc.dir/src/redirect.c.o reproc/CMakeFiles/reproc.dir/src/reproc.c.o reproc/CMakeFiles/reproc.dir/src/run.c.o reproc/CMakeFiles/reproc.dir/src/strv.c.o reproc/CMakeFiles/reproc.dir/src/utf.posix.c.o && /usr/bin/ranlib reproc/lib/libreproc.a && :
[16/18] /usr/bin/c++  -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/include -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c++11 -MD -MT reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o -MF reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o.d -o reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/src/reproc.cpp
FAILED: reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o 
/usr/bin/c++  -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/include -I/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc/include -fPIC -g -std=c++11 -MD -MT reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o -MF reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o.d -o reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o -c /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/src/reproc.cpp
In file included from /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/src/reproc.cpp:1:
/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/include/reproc++/reproc.hpp:95:5: error: declaration of ‘reproc::options::<unnamed struct> reproc::options::env’ changes meaning of ‘env’ [-Wchanges-meaning]
   95 |   } env = {};
      |     ^~~
/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/include/reproc++/reproc.hpp:91:5: note: used here to mean ‘class reproc::env’
   91 |     env::type behavior;
      |     ^~~
In file included from /home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/include/reproc++/reproc.hpp:11:
/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/include/reproc++/env.hpp:8:7: note: declared here
    8 | class env : public detail::array {
      |       ^~~
/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/include/reproc++/reproc.hpp:107:5: error: declaration of ‘reproc::options::<unnamed struct> reproc::options::redirect’ changes meaning of ‘redirect’ [-Wchanges-meaning]
  107 |   } redirect = {};
      |     ^~~~~~~~
/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/include/reproc++/reproc.hpp:102:5: note: used here to mean ‘struct reproc::redirect’
  102 |     redirect err;
      |     ^~~~~~~~
/home/siiky/projects/sonic-pi-4.4.0/app/vcpkg/buildtrees/reproc/src/v14.2.4-061178cb57.clean/reproc++/include/reproc++/reproc.hpp:68:8: note: declared here
   68 | struct redirect {
      |        ^~~~~~~~
ninja: build stopped: subcommand failed.
```
</details>

--- a/ports/reproc/fix-gcc13-build-error.patch
+++ b/ports/reproc/fix-gcc13-build-error.patch
@@ -1,0 +1,28 @@
+diff --git a/reproc++/include/reproc++/reproc.hpp b/reproc++/include/reproc++/reproc.hpp
+index ab6f139..d4b370f 100644
+--- a/reproc++/include/reproc++/reproc.hpp
++++ b/reproc++/include/reproc++/reproc.hpp
+@@ -88,18 +88,18 @@ struct redirect {
+ 
+ struct options {
+   struct {
+-    env::type behavior;
++    reproc::env::type behavior;
+     /*! Implicitly converts from any STL container of string pairs to the
+     environment format expected by `reproc_start`. */
+-    class env extra;
++    reproc::env extra;
+   } env = {};
+ 
+   const char *working_directory = nullptr;
+ 
+   struct {
+-    redirect in;
+-    redirect out;
+-    redirect err;
++    struct redirect in;
++    struct redirect out;
++    struct redirect err;
+     bool parent;
+     bool discard;
+     FILE *file;

--- a/ports/reproc/portfile.cmake
+++ b/ports/reproc/portfile.cmake
@@ -4,7 +4,8 @@ vcpkg_from_github(
     REF "v${VERSION}"
     SHA512 c592521960f1950d626261738091d25efdf764ee1a0c72a58c28c66eaebf6073b2c978f1dc2c8dbe89b0be7ec1629a3a45cb1fafa0ebe21b5df8d4d27c992675
     HEAD_REF main
-	PATCHES fix-gcc13-build-error.patch
+    PATCHES
+        fix-gcc13-build-error.patch #Please remove it in the next release version
 )
 
 vcpkg_cmake_configure(

--- a/ports/reproc/portfile.cmake
+++ b/ports/reproc/portfile.cmake
@@ -1,9 +1,10 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO DaanDeMeyer/reproc
-    REF v14.2.4
+    REF "v${VERSION}"
     SHA512 c592521960f1950d626261738091d25efdf764ee1a0c72a58c28c66eaebf6073b2c978f1dc2c8dbe89b0be7ec1629a3a45cb1fafa0ebe21b5df8d4d27c992675
     HEAD_REF main
+	PATCHES fix-gcc13-build-error.patch
 )
 
 vcpkg_cmake_configure(
@@ -24,8 +25,4 @@ foreach(TARGET reproc reproc++)
     )
 endforeach()
 
-file(
-    INSTALL "${SOURCE_PATH}/LICENSE"
-    DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
-    RENAME copyright
-)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

--- a/ports/reproc/vcpkg.json
+++ b/ports/reproc/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "reproc",
   "version": "14.2.4",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Cross-platform (C99/C++11) process library",
   "homepage": "https://github.com/DaanDeMeyer/reproc",
   "supports": "!uwp",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7398,7 +7398,7 @@
     },
     "reproc": {
       "baseline": "14.2.4",
-      "port-version": 1
+      "port-version": 2
     },
     "rest-rpc": {
       "baseline": "0.07",

--- a/versions/r-/reproc.json
+++ b/versions/r-/reproc.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "6fc63b3c029742441d22a288db9f37f5e016abbe",
+      "version": "14.2.4",
+      "port-version": 2
+    },
+    {
       "git-tree": "eeac14685480439ee52ac05ec68faebb2cbaffad",
       "version": "14.2.4",
       "port-version": 1

